angrmanagement/ui/views/disassembly_view.py

```diff
@@ -380,7 +375,11 @@ def display_function(self, function):
     def decompile_current_function(self):
 
         if self._current_function.am_obj is not None:
-            self.workspace.decompile_function(self._current_function.am_obj)
+            curr_ins = None
+            if self.infodock.selected_insns is not None:
+                curr_ins = next(iter(self.infodock.selected_insns))
```

Comment:
This is not the right fix. First of all, `self.infodock.selected_insns` is an `ObjectContainer`, so it will never be `None`. Second, the set wrapped inside `self.infodock.selected_insns` can be empty, which will cause `next(iter())` to fail with a `StopIteration`.

Reply:
Got it, my bad. Fixed.